src/mumble/UserLocalNicknameDialog.h

```diff
@@ -0,0 +1,36 @@
+// Copyright 2020 The Mumble Developers. All rights reserved.
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file at the root of the
+// Mumble source tree or at <https://www.mumble.info/LICENSE>.
+
+#ifndef MUMBLE_MUMBLE_USERNICKNAME_H_
+#define MUMBLE_MUMBLE_USERNICKNAME_H_
+
+#include <memory>
+
+#include "ClientUser.h"
+#include "ui_UserLocalNicknameDialog.h"
+
+class UserLocalNicknameDialog : public QDialog, private Ui::UserLocalNicknameDialog {
+	Q_OBJECT
+	Q_DISABLE_COPY(UserLocalNicknameDialog);
+
+	/// The session ID for the user that the dialog is changing the nickname for.
+	unsigned int m_clientSession;
+
+	/// The user's original nickname when entering the dialog.
+	QString m_originalNickname;
+	std::map< unsigned int, std::unique_ptr< UserLocalNicknameDialog > > &m_qmUserNicknameTracker;

```

Comment:
I just noticed that in C++ it is usually preferable to use `std::unordered_map` over `std::map` for things like we do here. Have a look at https://stackoverflow.com/a/13799886/3907364